src/shared/sleep-config.c

```diff
@@ -108,77 +111,132 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+/* Get the list of battery */
+static int get_sd_device_battery(sd_device_enumerator *e) {
```

Comment:
I do not think it is necessary to split the function like that.
At least this should take `sd_device_enumerator **e`. Otherwise, this function leaks memory, and the caller cannot obtain battery devices.

Reply:
Since it got used 3 times so i made a helper. To avoid memory leak, i have now passed address of pointer.